src/otx/engine/engine.py

```diff
@@ -0,0 +1,407 @@
+                ```
+        """
+        lit_module = self._build_lightning_module(
+            model=self.model,
+            optimizer=self.optimizer,
+            scheduler=self.scheduler,
+        )
+        if datamodule is None:
+            datamodule = self.datamodule
+        lit_module.meta_info = datamodule.meta_info
+
+        self._build_trainer(**kwargs)
+
+        self.trainer.test(
+            model=lit_module,
+            dataloaders=datamodule,
+            ckpt_path=str(checkpoint) if checkpoint is not None else self.checkpoint,
+        )
+
+        return self.trainer.callback_metrics
+
+    def predict(
+        self,
+        checkpoint: str | Path | None = None,
+        datamodule: EVAL_DATALOADERS | OTXDataModule | None = None,
+        return_predictions: bool | None = None,
+        **kwargs,
+    ) -> list | None:
+        """Run predictions using the specified model and data.
+
+        Args:
+            datamodule (EVAL_DATALOADERS | OTXDataModule | None, optional): The data module to use for predictions.
+            checkpoint (str | Path | None, optional): The path to the checkpoint file to load the model from.
+            return_predictions (bool | None, optional): Whether to return the predictions or not.
+            **kwargs: Additional keyword arguments for pl.Trainer configuration.
+
+        Returns:
+            list | None: The predictions if `return_predictions` is True, otherwise None.
+
+        Example:
+            >>> engine.predict(
+                datamodule=OTXDataModule(),
+                checkpoint="checkpoint.ckpt",
+                return_predictions=True,
+            )
+
+        CLI Usage:
+            1. you can pick a model.
+                ```python
+                otx predict
+                    --model <CONFIG | CLASS_PATH_OR_NAME> --data_root <DATASET_PATH, str>
+                    --checkpoint <CKPT_PATH, str>
+                ```
+            2. If you have a ready configuration file, run it like this.
+                ```python
+                otx predict --config <CONFIG_PATH, str> --checkpoint <CKPT_PATH, str>
+                ```
+        """
+        lit_module = self._build_lightning_module(
+            model=self.model,
+            optimizer=self.optimizer,
+            scheduler=self.scheduler,
+        )
+
+        self._build_trainer(**kwargs)
+
+        return self.trainer.predict(
+            model=lit_module,
+            datamodule=datamodule if datamodule is not None else self.datamodule,
+            ckpt_path=str(checkpoint) if checkpoint is not None else checkpoint,
+            return_predictions=return_predictions,
+        )
+
+    def export(self, *args, **kwargs) -> None:
+        """Export the trained model to OpenVINO Intermediate Representation (IR) or ONNX formats."""
+        raise NotImplementedError
+
+    """
+    Property and setter functions provided by Engine.
+    """
```

Comment:
Not sure if it is common to use triple quotes there. Let's stay on the safe side and stick to the conventional format :)

```suggestion
    # ------------------------------------------------------------------------ #
    # Property and setter functions provided by Engine.                        #
    # ------------------------------------------------------------------------ #
```

Reply:
Thanks! Done - https://github.com/openvinotoolkit/training_extensions/pull/2778/commits/367d9bc4dcbb549d9c065e7286abf587d1a36675